src/main/java/org/jabref/gui/util/ViewModelTextFieldTableCellVisualizationFactory.java

```diff
@@ -30,25 +38,65 @@ public void install(TableColumn<S, T> column, StringConverter<T> stringConverter
+        return new TextFieldTableCell<>(stringConverter) {
+
+            List<Subscription> subscriptions = new ArrayList<>();
+
+            @Override
+            public void startEdit() {
+                super.startEdit();
+
+                // The textfield is lazily created and not already present when a TableCell is created.
+                lookupTextField().ifPresent(textField -> Platform.runLater(() -> {
+                    textField.requestFocus();
+                    textField.selectAll();
+                }));
+            }
+
+            /**
+             * As 'textfield' is a private member of TextFieldTableCell we need need to get to it through the backdoor.
+             *
+             * @return The TextField containing the editable content of the TableCell
+             */
+            private Optional<TextField> lookupTextField() {
+                if (getGraphic() instanceof TextField) {
+                    return Optional.of((TextField) getGraphic());
+                } else {
+                    // Could be an HBox with some graphic and a TextField if a graphic is specified for the TableCell
+                    if (getGraphic() instanceof HBox) {
+                        HBox hbox = (HBox) getGraphic();
+                        if ((hbox.getChildren().size() > 1) && hbox.getChildren().get(1) instanceof TextField) {
+                            return Optional.of((TextField) hbox.getChildren().get(1));
+                        }
+                    }
+                    return Optional.empty();
+                }
+            }
 
             @Override
             public void updateItem(T item, boolean empty) {
                 super.updateItem(item, empty);
 
-                if (!empty && (getTableRow() != null)) {
-                    Object rowItem = getTableRow().getItem();
+                subscriptions.forEach(Subscription::unsubscribe);
+                subscriptions.clear();
 
-                    if (rowItem != null) {
-                        S vm = (S) rowItem;
-                        if ((visualizer != null) && (validationStatusProperty != null)) {
-                            visualizer.initVisualization(validationStatusProperty.apply(vm), this);
-                        }
+                S viewModel = getTableRow().getItem();
+                if (empty || (viewModel == null)) {
+                    setText(null);
+                    setGraphic(null);
+                    setOnMouseClicked(null);
+                    setTooltip(null);
+                    setStyle("");
+                } else {
+                    if (validationStatusProperty != null) {
+                        validationStatusProperty.apply(viewModel).getHighestMessage().ifPresent(message -> {
+                            setTooltip(new Tooltip(message.getMessage()));
+                            subscriptions.add(BindingsHelper.includePseudoClassWhen(this, INVALID_PSEUDO_CLASS,
```

Comment:
This line should probably be outside of the `isPresent` block as you want to subscribe to the `validProperty` in either case.